src/Core/Managed/Shared/TelemetryClient.cs

```diff
@@ -17,14 +17,19 @@
     /// <summary>
     /// Send events, metrics and other telemetry to the Application Insights service.
     /// </summary>
-    public sealed class TelemetryClient
+    public sealed class TelemetryClient : IDisposable
```

Comment:
I don't think appropriate use of the `IDisposable` interface is a problem. But why does the `TelemetryClient` need it now? @dnduffy Could you expand the description of this pull request to help us understand [what is the main change here and why is it made](http://chris.beams.io/posts/git-commit/#why-not-how)?